src/Tasks/SystemState.cs

```diff
@@ -573,6 +592,47 @@ private string[] GetDirectories(string path, string pattern)
             return getDirectories(path, pattern);
         }
 
+        /// <summary>
+        /// Cached implementation of GetFiles aimed to verify existence of a file in a directory.
+        /// It does not throw if directory do not exists.
+        /// </summary>
+        /// <param name="path"></param>
+        /// <param name="fileName"></param>
+        /// <returns>file full path or null if file do not exists</returns>
+        private string GetDirectoryFile(string path, string fileName)
```

Comment:
Is anything actually using the string return value? Can we save memory by changing the cache to `Dictionary<string, Dictionary<string, bool>>`?

Reply:
Originally I was using cached full path value, by callers, as opposed to Path.Combine it. But since that approach cause few unit tests failures (file name case sensitivity) I had to get back to original behavior.
I was hoping that we might revisit the need for that case sensitivity in unit tests, btu since this is not likely to happen, I have just rewritten the code to `Dictionary<string, HashSet<string>>`